server/src/main/java/com/google/sps/utils/AdRowProcessor.java

```diff
@@ -59,46 +67,138 @@ public static Ad convertRowToAd(String[] row) {
+    String[] impressionsArray = impressionsString.split("-");
+    return impressionsArray; 
+  }
+
+  /* 
+   * isTargetingAge determines whether or not the ad is targeting age.
+   */ 
+  public static boolean isTargetingAge(String str) {
+    if (str.trim().toLowerCase().equals("not targeted")) {
       return false; 
     }
     return true; 
   }
 
+  /* 
+   * checkGenderTarget determines if the arguments in the gender target field are valid.
+   * If they are not valid, the function will throw an exception.
+   */
+  public static List<String> checkGenderTarget(String str) throws IllegalArgumentException {
+    List<String> targets = convertStringToList(str);
+    Set<String> validGenderTargets = new HashSet<>(Arrays.asList("not targeted", "female", "male", "unknown gender"));
+    for (String target : targets) {
+      if (!validGenderTargets.contains(target.trim().toLowerCase())) {
+          throw new IllegalArgumentException("Gender target field contains invalid target.");
+      }
+    }
+    return targets;  
+  }
+
+  /* 
+   * checkGeoTargets determines if the arguments in the geo target field are valid.
+   * If they are not valid, the function will throw an exception.
+   */
+  public static List<String> checkGeoTarget(String str) throws IllegalArgumentException {
+    List<String> targets = convertStringToList(str);
+    Set<String> validGeoTargets = new HashSet<>(Arrays.asList("alabama", "alaska", "american samoa", "arizona", 
+                                                "arkansas", "california", "colorado", "connecticut", 
+                                                "delaware", "district of columbia", "florida", "georgia", 
+                                                "guam", "hawaii", "idaho", "illinois", "indiana", "iowa", 
+                                                "kansas", "kentucky", "louisiana", "maine", "maryland", 
+                                                "massachusetts", "michigan", "minnesota", "minor outlying islands", 
+                                                "mississippi", "missouri", "montana", "nebraska", "nevada", 
+                                                "new hampshire", "new jersey", "new mexico", "new york", 
+                                                "north carolina", "north dakota", "northern mariana islands", 
+                                                "ohio", "oklahoma", "oregon", "pennsylvania", "puerto rico", 
+                                                "rhode island", "south carolina", "south dakota", "tennessee", 
+                                                "texas", "u.s. virgin islands", "utah", "vermont", "virginia", 
+                                                "washington", "west virginia", "wisconsin", "wyoming", 
+                                                "united states", "the united states","not targeted"));
+    for (String target : targets) {
+      if (!validGeoTargets.contains(target.trim().toLowerCase())) {
```

Comment:
I see that you have the `.trim().toLowerCase()` repeated many times -> should we make a small function for that? 

ie. `trimAndLower(String str) { return .trim().toLowerCase(); }`